samples/bluetooth/mesh/src/main.c

```diff
@@ -373,6 +369,25 @@ static void button_pressed(struct k_work *work)
 
 	printk("Provisioned and configured!\n");
 }
+#endif
+
+static void button_pressed(struct k_work *work)
+{
+	bool is_provisioned = bt_mesh_is_provisioned();
+
+#ifdef CONFIG_BLE_MESH_SAMPLE_ENABLE_SELF_PROVISIONING
+	if (!is_provisioned) {
+		self_provision_device();
+		return;
+	}
+#endif
+
+	if (is_provisioned) {
+		(void)gen_onoff_send(!onoff.val);
+	} else {
+		printk("Must be provisioned before using!\n");
+	}
```

Comment:
Or it could be even:
```
static void button_pressed(struct k_work *work)
{
	if (bt_mesh_is_provisioned()) {
		(void)gen_onoff_send(!onoff.val);
		return;
	}
	else if (!IS_ENABLED(CONFIG_BLE_MESH_SAMPLE_ENABLE_SELF_PROVISIONING)) {
		printk("Must be provisioned before using!\n");
		return;
	}
```

Reply:
I've had it somewhat like that before to keep the impact low but then the static code checks complained about an "else after return statement". Wrapping the else in the "#ifndef" could fix it.

So if you're happy with moving the self provisioing in an own function then I'll try your first suggestion when I update this PR.

Reply:
Right, it can be simple `if` without `else`?
```
static void button_pressed(struct k_work *work)
{
	if (bt_mesh_is_provisioned()) {
		(void)gen_onoff_send(!onoff.val);
		return;
	}

	if (!IS_ENABLED(CONFIG_BLE_MESH_SAMPLE_ENABLE_SELF_PROVISIONING)) {
		printk("Must be provisioned before using!\n");
		return;
	}
```